app/validation/validator.py

```diff
@@ -881,14 +922,35 @@ def _get_answers_with_parent_ids(json_to_validate):
         for section in json_to_validate.get('sections'):
             for group in section.get('groups'):
                 for block in group.get('blocks'):
-                    if block.get('questions'):
-                        for question in block['questions']:
-                            for answer in question.get('answers', []):
-                                answers[answer['id']] = {
-                                    'answer': answer,
-                                    'block': block['id'],
-                                    'group_id': group['id'],
-                                    'section': section['id']
-                                }
+                    questions = []
+
+                    for variant in block.get('variants', []):
+                        if 'question' in variant:
+                            questions.append(variant.get('question', {}))
+
+                    single_question = block.get('question')
+                    if single_question:
+                        questions.append(single_question)
+
+                    for question in block.get('questions', []):
+                        questions.append(question)
+
+                    for question in questions:
+                        for answer in question.get('answers', []):
+                            answers[answer['id']] = {
+                                'answer': answer,
+                                'block': block['id'],
+                                'group_id': group['id'],
+                                'section': section['id']
+                            }
 
         return answers
+
+
+def find_duplicates(lists):
+    seen = set()
+    for lst in lists:
```

Comment:
😢 `list` not `lst`

Reply:
list is a builtin. I'll think of something else.